contributors/design-proposals/storage/volume-provisioning.md

```diff
@@ -451,6 +461,38 @@ will be extended to contain StorageClass.Parameters.
 
 The existing provisioner implementations will be modified to accept the StorageClass configuration object.
 
+### External provisioner interface changes
+
+Sometimes PersistentVolume, PersistentVolumeClaim or StorageClass objects get
+new fields and/or different behavior is expected from an external provisioner.
+Old provisioner that don't know about these new fields must not provision PVCs
+that expect the provisioner knows about them.
+
+As an example, in 1.8 we introduced `StorageClass.MountOptions` and
+`StorageClass.ReclaimPolicy`. Each provisioner must copy these fields from
+`StorageClass` to created PV during provisioning. Old provisioners do not know
+about this requirement and Kubernetes should not allow them to provision a PVC
```

Comment:
> Kubernetes should not allow them to provision a PVC

I assume you meant provisioning a PV here, but how does Kubernetes prevent a pre-1.8 external provisioner from creating the PV? How does Kubernetes know that an external provisioner exhibit the correct behavior for a given version of Kubernetes?

I understand why you're introducing the `volume.kubernetes.io/external-provisioning-version` annotation to the PVC, but an external provisioner can easily retrieve such information by contacting the API server. For example, [Trident](https://github.com/NetApp/trident) determines the Kubernetes version and decides whether to support the default storage class feature or not. As I said, the part that I don't understand is how Kubernetes enforces external provisioners to exhibit the correct behavior based on the version of Kubernetes. 

It would be helpful to document the changes related to provisioning/storage classes introduced in different versions of k8s on this page, so that we don't have to dig the CHANGELOG to know the expected behavior for each release. Thanks for copying me on the PR!